noir/bootstrap.sh

```diff
@@ -220,6 +220,10 @@ function bump_noir_repo_ref {
   git fetch --depth 1 origin $branch || true
   git checkout --track origin/$branch || git checkout $branch || git checkout -b $branch
   scripts/sync.sh write-noir-repo-ref $ref
+
+  # Build nargo and run formatter on `noir-projects`
+  build_native
+  ../noir-projects/bootstrap.sh format
   git add .
```

Comment:
Two thoughts:
* Doing this in the middle of the `bump-noir-ref` command seems like a hack. The command has well defined docs and expected behaviour, and I would not expect it to kick off a whole build process and add extra stuff to the commit. Can this not be done as a separate step in the workflow [after the bump](https://github.com/AztecProtocol/aztec-packages/blob/2c54bc181968d1769b0ced66c77d493aa34bee14/.github/workflows/pull-noir.yml#L66-L72)?
* I'm not sure what ensures that the `$NARGO` variable that the `format` commands above are looking for is going to point at the `target` directory.

Reply:
- My understanding around the `bump-noir-ref` command is that it was intentionally chosen to do everything necessary for bumping the noir ref. If we split this logic out then we'd be making it harder to have consistency between CI and the CLI.
- `$NARGO` is set to point at the target directory of the noir repo by default.

Reply:
Ah, I thought it was just made to encapsulate all the git commands to do the bumping itself. There could be other things it breaks, like the smart contracts themselves. Formatting is something that can be automated, but it's unlikely we can cover everything that can go wrong.

@ludamad suggested this command, so I'll leave it up to him to tell what he thought its scope should be.

Reply:
Agreed that we can't automate away every change necessary for noir syncs to go through. Formatting changes have been raised as a pain point however so I'm doing what I can.

Reply:
It seems pull-noir.yml is not exactly set up to do a build though - unless we're ok with it just happening on a github runner

Reply:
but yes I still very much believe that we should have all logic testable thru CI, the more we accumulate workflows that do non-trivial things the harder it is for people to ask people to make sure those work (and pain for the people who have to run them)

Reply:
Yep, I think that the slowdown from doing a noir build is acceptable considering that most runs of this command are done by a bot rather than a human. If we run into troubles then we can split this into "bump" and "bump and fixup" commands.